api/src/resources/source.rs

```diff
@@ -123,6 +137,9 @@ pub struct NewSource<'request> {
 
     #[serde(skip_serializing_if = "Option::is_none")]
     pub sensitive_properties: Option<Vec<&'request str>>,
+
+    #[serde(skip_serializing_if = "Option::is_none", rename = "_type")]
+    pub source_type: Option<&'request str>,
```

Comment:
:+1: on `rename`

I think you should use the newtype here though, e.g. like `BucketId` -- although, you'd want to borrow it rather than copy it 
```suggestion
    pub source_type: Option<&'request SourceType>,
```

Reply:
Ok!